src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/CreateGcsBucketStep.java

```diff
@@ -61,14 +74,22 @@ public StepResult doStep(FlightContext flightContext)
             .setLifecycleRules(ApiConversions.toGcsApi(creationParameters.getLifecycle()))
             .build();
 
-    final StorageCow storageCow = crlService.createStorageCow(userRequest);
+    final StorageCow storageCow = crlService.createStorageCow(gcpProjectId, userRequest);
     storageCow.create(bucketInfo);
     return StepResult.getStepResultSuccess();
   }
 
   @Override
   public StepResult undoStep(FlightContext flightContext) throws InterruptedException {
-    final StorageCow storageCow = crlService.createStorageCow(userRequest);
+    String gcpProjectId =
+        workspaceDao
+            .getGcpCloudContext(resource.getWorkspaceId())
+            .orElseThrow(
```

Comment:
Usually the style is to `.map(GcpCloudContext::getGcpProjectId).orElseThrow(...)`. I've always seen the throw come last.

Reply:
you're right, that feels neater